src/ukrdc_cupid/core/store/delete_old.py

```diff
@@ -0,0 +1,132 @@
+import ukrdc_sqla.ukrdc as sqla
+from sqlalchemy.orm import Session
+from sqlalchemy import and_
+from typing import List, Optional
+
+
+def delete_records(
+    table: sqla.Base,
+    session: Session,
+    pid: Optional[str],
+    exclude_ids: Optional[List[str]],
+):
+    """Generic function to delete a set of records from there host table based on the pid and a list of ids.
+
+    Args:
+        table (sqla.Base): _description_
+        session (Session): _description_
+        pid (str): _description_
+        exclude_ids (Optional[List[str]]): _description_
+    """
+
+    if exclude_ids is not None and pid is None:
+        records_to_delete = (
+            session.query(table)
+            .where(and_(table.id.not_in(exclude_ids), table.pid == pid))
+            .all()
+        )
+    elif exclude_ids is None and pid is not None:
+        records_to_delete = (
+            session.query(table)
+            .where(
+                and_(
+                    table.id.not_in(exclude_ids),
+                )
+            )
+            .all()
+        )
+    else:
+        records_to_delete = session.query(table).where(table.pid == pid).all()
+
+    for record in records_to_delete:
+        session.delete(record)
+
+
+def delete_timebound_records(
+    table: sqla.Base,
+    session: Session,
+    pid: Optional[str],
+    exclude_ids: Optional[List[str]],
+    start,
+    stop,
+):
+    return
+
+
+# a few ugly duckings need special handling
+def delete_result_items():
+    return
+
+
+def delete_survey_children():
+    return
+
+
+# Map delete function to table name
+SQL_DELETE = {
```

Comment:
This instinctively makes me uncomfortable but I'm curious to see how it goes... By all means carry on with it, I'm just interested to see how it ends up working/looking.

Reply:
This is redundant for now. I haven't cleaned up all the files with _old after them as a note to myself that they may contain things I want to reinstate.